firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/FirebaseAppDistributionTest.java

```diff
@@ -452,20 +453,36 @@ public void updateToNewRelease_receiveProgressUpdateFromUpdateApp() {
   }
 
   @Test
-  public void taskCancelledOnScreenRotation_whenUpdateDialogShowing() {
+  public void signInConfirmationDialogReappears_whenScreenRotates_whenSignInDialogShowing() {
+    when(mockSignInStorage.getSignInStatus()).thenReturn(false);
+    when(activity.isChangingConfigurations()).thenReturn(true);
+
+    UpdateTask updateTask = firebaseAppDistribution.updateIfNewReleaseAvailable();
+
+    // Mimic activity dying
+    firebaseAppDistribution.onActivityDestroyed(activity);
+    firebaseAppDistribution.onActivityCreated(activity);
+
+    assertAlertDialogShown();
+  }
+
+  @Test
+  public void updateDialogReappears_whenScreenRotates_whenSignInDialogShowing() {
     AppDistributionReleaseInternal newRelease = TEST_RELEASE_NEWER_AAB_INTERNAL.build();
     when(mockNewReleaseFetcher.checkForNewRelease()).thenReturn(Tasks.forResult(newRelease));
+    when(activity.isChangingConfigurations()).thenReturn(true);
 
     UpdateTask updateTask = firebaseAppDistribution.updateIfNewReleaseAvailable();
 
     // Mimic activity dying
     firebaseAppDistribution.onActivityDestroyed(activity);
+    firebaseAppDistribution.onActivityCreated(activity);
 
-    assertTaskFailure(updateTask, INSTALLATION_CANCELED, UPDATE_CANCELED);
+    assertAlertDialogShown();
```

Comment:
Oh, can you also add assertions on these tests checking that the task does not get canceled?

Reply:
will do!